### PR TITLE
Grin vs BEAM update: BEAM clarifications 

### PR DESCRIPTION
**Pull request by philipr-za**
The BEAM CEO (A Zaidelson) sent through some clarifications on the Grin vs BEAM comparison (can be found in https://github.com/tari-labs/tari-university/issues/67).

I have incorporated the updates.

Unfortunately my Markdown editor changed the word wrap on the whole document so the changes will be hard to spot but they were very minor.

### CODE REVIEW

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 1]
```diff
@@ -35,78 +38,83 @@ Both Grin and BEAM have incorporated a version of the Dandelion relay protocol t
...
+Grin has implemented a method for a node to sync the block chain very quickly by only downloading a partial history [[11]]. A new node entering the network will query the current head block of the chain and then requests the block header at a horizon, in the example the horizon is initially set at 5000 blocks before the current head. The node then checks if there is enough data to confirm consensus and if there isn't it will increase its horizon until consensus is reached. At that point it will download the full UTXO set of the horizon block. This approach does introduce a few security risks but mitigations are provided and the result is that a node can sync to the network with an order of magnitude less data.
+
+**Update 24-12-2018**: Since the initial writing of this article BEAM has published their solution for fast node synchronization using macroblocks. A macroblock is a complete state of all UTXOs, periodically created by Beam nodes [[12]].
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Suggest change 1st sentence as follows: "_Since the initial writing of this article (October 2018) BEAM has published their solution for fast node synchronization using macroblocks._"

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 2]
```diff
@@ -35,78 +38,83 @@ Both Grin and BEAM have incorporated a version of the Dandelion relay protocol t
...
+When constructing a valid Mimblewimble transaction the parties involved need to collaborate in order to choose blinding factors that balance. This interactive negotiation requires a number of steps and it implies that the parties need to be in communication to finalize the transaction. Grin facilitates this process by the two parties connecting directly to one another using a socket based channel for a "real-time" session. This means that both parties need to be online simultaneously. BEAM has implemented a secure Bulletin Board System (SBBS) system that is run on BEAM full-nodes to allow for asynchronous negotiation of transactions [[30]], [[31]].
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Change 
"_implemented a secure Bulletin Board System (SBBS) system that is run_" 
to
"_implemented a Secure Bulletin Board System (SBBS) that is run_"

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 3]
```diff
@@ -35,78 +38,83 @@ Both Grin and BEAM have incorporated a version of the Dandelion relay protocol t
...
+Requiring the interactive participation of both parties in constructing a transaction can be a point of friction in using a Mimblewimble block chain. In addition to the secure BBS communication channel BEAM also plans to supports one-sided transactions where the payee in a transaction who expects to be paid a certain amount can construct their half of the transaction and send this half constructed transaction to the payer. The payer can then finish constructing the transaction and publish it to the block chain. Under the normal Mimblewimble system this is not possible because it would involve revealing your blinding factor to the counterparty. BEAM solves this problem by using a process they call *kernel fusion* whereby a kernel can include a reference to another kernel so that it is only valid if both kernels are present in the transaction. In this way the payee can build their half of the transaction with a secret blinding factor and a kernel that compensates for their blinding factor that **must** be included when the payer completes the transaction [[13]]. BEAM has indicated that this feature will be part of the initial release.
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Change "_BEAM also plans to supports one-sided transactions_" to "_BEAM also plans to support one-sided transactions_"

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 4]
```diff
@@ -1,32 +1,35 @@
...
+
+The two projects are being independently built from scratch by different teams in different languages (Rust [[6]] and C++ [[7]]), so there will be many differences in the raw implementations. For example, Grin uses LMDB for its embedded SQL database and BEAM use SQLite which have performance differences but are functionally similar. Grin uses a Directed Acyclic Graph (DAG) to represent their mempool to avoid transaction reference loops [[8]] while BEAM uses a multiset key-value data structure with logic to enable some of their extended features [[7]].
 
 From the perspective of features the two projects exhibit all the features inherent to Mimblewimble. Grin's stated goal is to produce a simple and easy to maintain implementation of the Mimblewimble protocol [[3]]. BEAM's implementation however contains a number of modifications to the Mimblewimble approach with the aim to provide some unique features for their implementation. Before we get into the features and design elements that set the two project apart let's discuss an interesting feature that both projects have implemented.
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Change "_that set the two project apart_" to "_that set the two projects apart_"

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 5]
```diff
@@ -1,32 +1,35 @@
 # Grin vs. BEAM, a Comparison
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Change all "_block chain_" to "_blockchain_"

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 6]
```diff
@@ -1,32 +1,35 @@
...
 
 ## Introduction
+
 Grin and BEAM are two open-source cryptocurrency projects based on the Mimblewimble protocol. The Mimblewimble protocol was first proposed by a anonymous user using the pseudonym Tom Elvis Jedusor (the french translation of Voldemort's name from the Harry Potter series of books). This user logged onto a bitcoin research IRC channel and posted a link to a text article hosted on a Tor hidden service [[1]]. This article provided the basis for a new way to construct block chain style transactions that provided inherent privacy and the ability to dramatically reduce the size of the block chain by compressing the transaction history of the chain. This initial article presented the main ideas of the protocol but it left out a number of critical elements required for a practical implementation and even contained a mistake in the cryptographic formulation. Andrew Poelstra published a follow-up paper that addresses many of these issues and refines the core concepts of Mimblewimble [[2]] which have been applied to the practical implementations of this protocol in both the Grin [[3]] and BEAM [[4]] projects.
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Need comma before but in "_main ideas of the protocol but it left out_"

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 7]
```diff
@@ -35,78 +38,83 @@ Both Grin and BEAM have incorporated a version of the Dandelion relay protocol t
...
...m with the parameters set to n=150, k=5 [[32]]. Equihash was proposed in 2016 as a memory-hard PoW algorithm which relied heavily on memory-usage to achieve Application Specific Integrated Circuit (ASIC) resistance [[19]]. The goal was to produce an algorithm that would be more efficient to run on consumer GPUs as opposed to the growing field of ASIC miners, mainly produced by Bitmain at the time. It was hoped this would aid in decentralising the mining power for cryptocurrencies that used this algorithm. The idea behind Equihash's ASIC resistance was that at the time implementing memory in an ASIC was expensive and so GPUs were more efficient at calculating the Equihash PoW. This ASIC resistance did last for a while but in early 2018 Bitmain released an ASIC for Equihash which were significantly more efficient than GPUs for the Equihash configurations used by Zcash, Bitcoin Gold and Zencash to name a few. It is possible to tweak the parameters of the Equihash algorithm to make it more memory intensive and thus make current ASICs and the older GPU mining farms obsolete but it remains to be seen if BEAM will do this. No block time has been published as of the writing of this report.
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Need comma before but in "_and the older GPU mining farms obsolete but it remains_"

File: src/protocols/grin-beam-comparison/MainReport.md [hunk 8]
```diff
@@ -35,78 +38,83 @@ Both Grin and BEAM have incorporated a version of the Dandelion relay protocol t
...
+In terms of the monetary policy of the two projects BEAM has stated that they will be using a deflationary model with periodic halving of their mining reward and a maximum supply of BEAM of ~262 million coins. Beam will start with 100 coins emitted per block. The first halving will occur after one year, and then halving will happen every 4 years [[32]]. Grin has opted for an inflationary model where the block reward will remain constant, they make their arguments for this approach in [[27]]. This approach will asymptotically tend towards a zero percent dilution as the supply increases instead of enforcing a set supply [[28]]. Grin has not specified their mining reward or fees structure as yet but based on their current documentation Grin is planning on a 60 Grin per block reward. Neither project has made a final decision of how to structure fees but the Grin project has started to explore how to set a fee baseline by using a metric of "fees per reward per minute" [[28]].
```
Comment (hansieodendaal) [inline src/protocols/grin-beam-comparison/MainReport.md]: Need comma before but in "_made a final decision of how to structure fees but the Grin project has_"

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 1 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: Done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 2 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: Done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 3 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: Done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 5 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: Done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 4 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 6 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 7 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 8 above)
Comment (philipr-za) [reply src/protocols/grin-beam-comparison/MainReport.md]: Done

File: src/protocols/grin-beam-comparison/MainReport.md (hunk 1 above)
Comment (hansieodendaal) [reply src/protocols/grin-beam-comparison/MainReport.md]: Please also remove "**Update 24-12-2018**: "

**Pull request merged**